lib/matplotlib/tests/test_table.py

```diff
@@ -229,3 +233,30 @@ def test_table_bbox(fig_test, fig_ref):
                   loc='center',
                   bbox=Bbox.from_extents(0.1, 0.2, 0.9, 0.8)
                   )
+
+
+@check_figures_equal()
+def test_table_unit(fig_test, fig_ref):
+    # test that table doesn't participate in unit machinery, instead uses repr/str
+
+    class FakeUnit:
+        def __init__(self, thing):
+            pass
+        def __repr__(self):
+            return "Hello"
+
+    fake_convertor = munits.ConversionInterface()
+    fake_convertor.convert = Mock(side_effect=lambda v, u, a: 0)
+    # not used, here for completeness
+    fake_convertor.default_units = Mock(side_effect=lambda v, a: None)
+    fake_convertor.axisinfo = Mock(side_effect=lambda u, a: munits.AxisInfo())
+
+    munits.registry[FakeUnit] = fake_convertor
+
+    data = [[FakeUnit("yellow"), FakeUnit(42)],
+            [FakeUnit(datetime.datetime(1968, 8, 1)), FakeUnit(True)]]
+
+    fig_test.subplots().table(data)
+    fig_ref.subplots().table([["Hello", "Hello"], ["Hello", "Hello"]])
+    fig_test.canvas.draw()
+    assert not fake_convertor.convert.called
```

Comment:
```suggestion
    fake_convertor.convert.assert_not_called()
```

Also, if you are specifically targeting the datetime converter I _think_ you can save a few lines by having mock wrap the real converter's method rather than creating a new mocked object.  I don't know anything about the converters but I have an example of mock wrapping a method [here](https://github.com/matplotlib/matplotlib/pull/26311/files#diff-56298727b96c2c05776fa59f72ae5f189bb16d046d5452cf8379a1fe55906bdcR250-R254).